src/Streamly/Internal/Data/Stream/Async/Channel.hs

```diff
@@ -163,7 +164,10 @@ appendGeneric newChan modifier stream1 stream2 = K.concatEffect action
 {-# INLINE appendWithK #-}
 appendWithK :: MonadAsync m =>
     (Config -> Config) -> K.Stream m a -> K.Stream m a -> K.Stream m a
-appendWithK = appendGeneric Append.newChannel
+appendWithK modifier stream1 stream2 =
+    if getOrdered (modifier defaultConfig)
+    then concatMapWithK modifier id (stream1 `K.cons` K.fromPure stream2)
+    else appendGeneric Append.newChannel modifier stream1 stream2
```

Comment:
When the `ordered` is not set in the modifier are both the branches here same?

Reply:
If so, is `appendGeneric` more performant? Hench the `if` statement exists?

Reply:
Checked performance, it is the same. We can use concat itself for both cases. But in future if we want to use StreamD for fusion, the special append impl may be useful.